packages/flutter/lib/src/rendering/object.dart

```diff
@@ -3609,14 +3610,34 @@ class _SemanticsGeometry {
   static final Matrix4 _temporaryTransformHolder = Matrix4.zero();
 
   /// From parent to child coordinate system.
-  static Rect _transformRect(Rect rect, RenderObject parent, RenderObject child) {
+  static Rect _transformRect(Rect rect, Matrix4 transform) {
+    assert(transform != null);
     if (rect == null)
       return null;
     if (rect.isEmpty)
       return Rect.zero;
-    _temporaryTransformHolder.setIdentity();  // clears data from a previous call
-    parent.applyPaintTransform(child, _temporaryTransformHolder);
-    return MatrixUtils.inverseTransformRect(_temporaryTransformHolder, rect);
+    return MatrixUtils.inverseTransformRect(transform, rect);
+  }
+
+  static void _applyIntermediatePaintTransforms(
+    RenderObject ancestor,
+    RenderObject child,
+    Matrix4 transform,
+    Matrix4 clipRectTransform,
```

Comment:
Alternatively, instead of moving `setIdentity` here in my suggestion above, we could assert that the matrix `isIdentity` here.

Reply:
I prefer that, done.